controllers/awsmachine_controller_test.go

```diff
@@ -1076,3 +1458,119 @@ func expectConditions(g *WithT, m *infrav1.AWSMachine, expected []conditionAsser
+				Values: aws.StringSlice([]string{"test"}),
+			},
+			{
+				Name:   aws.String("instance-state-name"),
+				Values: aws.StringSlice([]string{"pending", "running"}),
+			},
+		},
+	})).Return(&ec2.DescribeInstancesOutput{}, nil)
+	m.DescribeImages(gomock.Eq(&ec2.DescribeImagesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("owner-id"),
+				Values: aws.StringSlice([]string{"258751437250"}),
+			},
+			{
+				Name:   aws.String("name"),
+				Values: aws.StringSlice([]string{"capa-ami-ubuntu-18.04-?test-*"}),
+			},
+			{
+				Name:   aws.String("architecture"),
+				Values: aws.StringSlice([]string{"x86_64"}),
+			},
+			{
+				Name:   aws.String("state"),
+				Values: aws.StringSlice([]string{"available"}),
+			},
+			{
+				Name:   aws.String("virtualization-type"),
+				Values: aws.StringSlice([]string{"hvm"}),
+			},
+		}})).Return(&ec2.DescribeImagesOutput{Images: []*ec2.Image{
+		{
+			ImageId:      aws.String("latest"),
+			CreationDate: aws.String("2019-02-08T17:02:31.000Z"),
+		},
+	}}, nil)
+	m.RunInstances(gomock.Any()).Return(&ec2.Reservation{
+		Instances: []*ec2.Instance{
+			{
+				State: &ec2.InstanceState{
+					Name: aws.String(ec2.InstanceStateNameRunning),
+				},
+				IamInstanceProfile: &ec2.IamInstanceProfile{
+					Arn: aws.String("arn:aws:iam::123456789012:instance-profile/foo"),
+				},
+				InstanceId:     aws.String("two"),
+				InstanceType:   aws.String("m5.large"),
+				SubnetId:       aws.String("subnet-1"),
+				ImageId:        aws.String("ami-1"),
+				RootDeviceName: aws.String("device-1"),
+				BlockDeviceMappings: []*ec2.InstanceBlockDeviceMapping{
+					{
+						DeviceName: aws.String("device-1"),
+						Ebs: &ec2.EbsInstanceBlockDevice{
+							VolumeId: aws.String("volume-1"),
+						},
+					},
+				},
+				Placement: &ec2.Placement{
+					AvailabilityZone: aws.String("us-east-1a"),
+				},
+			},
+		},
+	}, nil)
+	m.WaitUntilInstanceRunningWithContext(gomock.Any(), gomock.Any(), gomock.Any()).
+		Return(nil)
+	m.DescribeNetworkInterfaces(gomock.Eq(&ec2.DescribeNetworkInterfacesInput{Filters: []*ec2.Filter{
+		{
+			Name:   aws.String("attachment.instance-id"),
+			Values: aws.StringSlice([]string{"two"}),
+		},
+	}})).Return(&ec2.DescribeNetworkInterfacesOutput{
+		NetworkInterfaces: []*ec2.NetworkInterface{
+			{
+				NetworkInterfaceId: aws.String("eni-1"),
+				Groups: []*ec2.GroupIdentifier{
+					{
+						GroupId: aws.String("3"),
+					},
+				},
+			},
+		}}, nil).MaxTimes(2)
+	m.DescribeNetworkInterfaceAttribute(gomock.Eq(&ec2.DescribeNetworkInterfaceAttributeInput{
+		NetworkInterfaceId: aws.String("eni-1"),
+		Attribute:          aws.String("groupSet"),
+	})).Return(&ec2.DescribeNetworkInterfaceAttributeOutput{Groups: []*ec2.GroupIdentifier{{GroupId: aws.String("3")}}}, nil).MaxTimes(1)
+	m.ModifyNetworkInterfaceAttribute(gomock.Eq(&ec2.ModifyNetworkInterfaceAttributeInput{NetworkInterfaceId: aws.String("eni-1"), Groups: aws.StringSlice([]string{"3", "1", "2"})})).AnyTimes()
```

Comment:
Why do we  mock `ModifyNetworkInterfaceAttribute()` twice, when we mentioned `AnyTimes()`. How does ordering of slice elements matters ? 🤔

Reply:
The order of slice is coming different every time, I will try to sort the elements.

Reply:
Done